src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java

```diff
@@ -1717,6 +1718,7 @@ public void applyConfig(Configuration config, boolean reindex, boolean interacti
         getAuthorizationFramework().reload();
 
         messagesContainer.setMessageLimit(config.getMessageLimit());
+        SuggesterServiceImpl.getInstance().refresh(configuration);
```

Comment:
Only thing I've got here is that perhaps when you use this approach of design (interface - implementaion) then the implementation should never occur anywhere in the code except for the constructing.

So maybe a factory, or factory method would be more appropriate and the suggester service would be a part of `RuntimeEnvironment`? The point is that the concrete implementation is always hidden to the end user.

If it cannot be done, leave it like this.

Reply:
I use the interface-implementation contract in Jersey (`SuggesterAppBinder`) which allows me to do dependency injection (`@Inject`). I cannot use it here since `RuntimeEnvironment` is used in `indexer` :/ I did not know what would be the best solution. At first, I started only using it in the `suggester` context but then I realized that I need to call it from some other parts as well... 

In the near future, after mavenization is in place, I would like to divide and move relevant parts to `indexer` and `web` modules.

However, still `SuggesterService.getInstance()` would be better (but it's far from what I would prefer) :)

Reply:
So what about `SuggesterServiceFactory.getDefault()` or something like that?

Problem with `SuggesterService.getInstance()` is that it is an interface.

Reply:
What do you mean? You can have `static` methods in Java 8.

Reply:
Yes but this is a place for a factory not a factory method of an interface. It's definitely better than now but still a way to improve.

Reply:
Done :)